t/generic.t

```diff
@@ -217,3 +217,8 @@ $old = $foo->query("q");
 print "not " unless !defined($old) && $foo eq "?q";
 print "ok 48\n";
 
+# canonical must always be a clone
+my $c1 = $foo->canonical; # canonicalize first
+my $c2 = $c1->canonical;  # canonicalize again
+print 'not ' if $c1 == $c2;
```

Comment:
I'm not sure if you're intending to compare refaddrs here or the underlying contents, but there is a `==` overload on the object that effectively does the latter.  To truly check that the object was cloned (has a different memory address), use `refaddr` from Scalar::Util.

Best would be to test both: that the contents are the same, and the address changed.